lib/core/test/unit/Cardano/WalletSpec.hs

```diff
@@ -362,7 +362,8 @@ walletListTransactionsSorted wallet@(wid, _, _) history =
     monadicIO $ liftIO $ do
         (WalletLayerFixture db wl _ slotIdTime) <- liftIO $ setupFixture wallet
         unsafeRunExceptT $ putTxHistory db (PrimaryKey wid) history
-        txs <- unsafeRunExceptT $ listTransactions wl wid
+        txs <- unsafeRunExceptT $
+            listTransactions wl wid Nothing Nothing Nothing
```

Comment:
The function fingerprint doesn't really read well at the call-site with the triple-Maybe. Would it be worth considering an intermediate type:

```hs
data Pagination = Pagination
    { start :: Maybe UTCTime
    , end :: Maybe UTCTime
    ,  direction :: SortOrder
    }

defaultPagination :: Pagination
defaultPagination = Pagination Nothing Nothing Ascending
```

@jonathanknowles ? What do you think? That would read a bit better in a few places IMO.

Reply:
> The function fingerprint doesn't really read well at the call-site with the triple-Maybe. Would it be worth considering an intermediate type:
> 
> ```haskell
> data Pagination = Pagination
>     { start :: Maybe UTCTime
>     , end :: Maybe UTCTime
>     ,  direction :: SortOrder
>     }
> 
> defaultPagination :: Pagination
> defaultPagination = Pagination Nothing Nothing Ascending
> ```
> 
> @jonathanknowles ? What do you think? That would read a bit better in a few places IMO.

That sounds very reasonable. I'll make another PR with this.